fdbserver/networktest.actor.cpp

```diff
@@ -57,22 +104,60 @@ ACTOR Future<Void> networkTestServer() {
 	}
 }
 
-ACTOR Future<Void> testClient( std::vector<NetworkTestInterface> interfs, int* sent ) {
-	loop {
-		NetworkTestReply rep = wait(  retryBrokenPromise(interfs[deterministicRandom()->randomInt(0, interfs.size())].test, NetworkTestRequest( LiteralStringRef("."), FLOW_KNOBS->NETWORK_TEST_REPLY_SIZE ) ) );
+static bool more_requests_pending(int count) {
+	if (count == -1) {
+		return false;
+	} else {
+		int request_count = FLOW_KNOBS->NETWORK_TEST_REQUEST_COUNT;
+		return (!request_count) || count < request_count;
+	}
+}
+
+static bool more_logging_needed(int count, int iteration) {
+	if (FLOW_KNOBS->NETWORK_TEST_SCRIPT_MODE) {
+		return iteration <= 2;
+	} else {
+		return more_requests_pending(count);
+	}
+}
+
+ACTOR Future<Void> testClient( std::vector<NetworkTestInterface> interfs, int* sent, int* completed, latency_stats * latency ) {
+	state std::string request_payload( FLOW_KNOBS->NETWORK_TEST_REQUEST_SIZE, '.');
+	state int count = FLOW_KNOBS->NETWORK_TEST_REQUEST_COUNT;
+	state latency_stats::sample sample;
+
+	while (more_requests_pending(*sent)) {
 		(*sent)++;
+		sample = latency->tick();
+		NetworkTestReply rep = wait(  retryBrokenPromise(interfs[deterministicRandom()->randomInt(0, interfs.size())].test, NetworkTestRequest( StringRef(request_payload), FLOW_KNOBS->NETWORK_TEST_REPLY_SIZE ) ) );
+		latency->tock(sample);
+		(*completed)++;
 	}
+	return Void();
 }
 
-ACTOR Future<Void> logger( int* sent ) {
+ACTOR Future<Void> logger( int * sent, int* completed, latency_stats * latency ) {
```

Comment:
There is a `clang-format` config file, so I'd recommend getting and running [git clang-format](https://github.com/llvm-mirror/clang/blob/master/tools/clang-format/git-clang-format), so that you don't have to think about where spaces go around `*` in type expressions.

Reply:
thanks!